src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,116 @@
+import Boom from 'boom';
+import { get, omit } from 'lodash';
+
+import {
+  createFindQuery,
+  createFilterPath,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, options = {}) {
+    const body = omit(options, 'id');
+
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return Object.assign({
+      type: response._type,
+      id: response._id,
+      version: response._version
```

Comment:
I think we should decide between:

1. maintain es style separation, maybe rename `_source` to something less search-engine-y
    ```json
    {
      "id": "123",
      "type" : "index-pattern",
      "version": 1,
      "attributes|doc|source|fields|values|whatever": {
        ...
      }
    }
    ```
1. Formally blacklist reserved field names, perhaps by reserving the `_` prefix (it's better to reserve now rather than down the road), throw errors when they are sent as part of the body, prevent ambiguity by not allowing:
    ```
    PUT /api/saved_objects/index-pattern/123
    {"type": "new-type","id":"new-id"}
    ```